src/util/memory.h

```diff
@@ -33,7 +33,7 @@
 
 // If user hasn't specified COMPILER_SUPPORTS_MAKE_UNIQUE then try to figure out
 // based on compiler version if std::make_unique is provided.
-#if defined(COMPILER_SUPPORTS_MAKE_UNIQUE)
+#if !defined(COMPILER_SUPPORTS_MAKE_UNIQUE)
```

Comment:
Whoops indeed. My testing had been more focused around compilers that didn't have std::make_unique and less around the compilers that already did.

In addition, yes, the gcc (and clang?) checks should also check __cplusplus in order to differentiate between -std=c++98 or c++11 or c++14.